File: hrms-backend/src/main/java/com/hrms/service/EventService.java

//service/EventService.java
package com.hrms.service;

import com.hrms.dto.request.EventRequest;
import com.hrms.dto.response.EventResponse;

import java.time.LocalDateTime;
import java.util.List;

public interface EventService {
 EventResponse create(EventRequest request, Long userId);
 EventResponse update(Long id, EventRequest request);
 EventResponse getById(Long id, Long employeeId);
 List<EventResponse> getAll();
 List<EventResponse> getUpcoming();
 List<EventResponse> getByDateRange(LocalDateTime start, LocalDateTime end);
 List<EventResponse> getForEmployee(Long employeeId);
 void delete(Long id);
 EventResponse updateStatus(Long id, String status);
 void updateRsvp(Long eventId, Long employeeId, String status);
}

File: hrms-frontend/src/services/eventService.js
// src/services/eventService.js
import api from '../api/axiosConfig';

const eventService = {
  getAll: () => api.get('/events'),
  getUpcoming: () => api.get('/events/upcoming'),
  getForEmployee: (empId) => api.get(`/events/employee/${empId}`),
  getByDateRange: (start, end) =>
    api.get(`/events/date-range?start=${start}&end=${end}`),
  getById: (id, empId) =>
    api.get(`/events/${id}${empId ? `?employeeId=${empId}` : ''}`),
  create: (data) => api.post('/events', data),
  update: (id, data) => api.put(`/events/${id}`, data),
  delete: (id) => api.delete(`/events/${id}`),
  updateStatus: (id, status) =>
    api.patch(`/events/${id}/status?status=${status}`),
  updateRsvp: (eventId, data) =>
    api.post(`/events/${eventId}/rsvp`, data),
};

export default eventService;